src/serena/cli.py

```diff
@@ -424,16 +424,33 @@ def index(project: str, log_level: str = "WARNING") -> None:
     @click.command("index-deprecated", help="Deprecated alias for 'serena project index'.")
     @click.argument("project", type=click.Path(exists=True), default=os.getcwd(), required=False)
     @click.option("--log-level", type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]), default="WARNING")
-    def index_deprecated(project: str, log_level: str = "WARNING") -> None:
+    @click.option("--max-memory", type=int, help="Maximum memory (MB) for Intelephense language server (minimum 256)")
```

Comment:
we shouldn't have language specific things here, but you can add general `--ls-max-memory`